src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -131,26 +131,41 @@ private TestSuite Build(Assembly assembly, string assemblyPath, IDictionary<stri
                 if (options.ContainsKey(FrameworkPackageSettings.DefaultTestNamePattern))
                     TestNameGenerator.DefaultTestNamePattern = options[FrameworkPackageSettings.DefaultTestNamePattern] as string;
 
-                if (options.ContainsKey(FrameworkPackageSettings.TestParameters))
+                if (options.ContainsKey(FrameworkPackageSettings.TestParametersDictionary))
                 {
-                    string parameters = options[FrameworkPackageSettings.TestParameters] as string;
-                    if (!string.IsNullOrEmpty(parameters))
-                        foreach (string param in parameters.Split(new[] { ';' }))
-                        {
-                            int eq = param.IndexOf("=");
+                    var testParametersDictionary = options[FrameworkPackageSettings.TestParametersDictionary] as IDictionary<string, string>;
+                    if (testParametersDictionary != null)
+                    {
+                        foreach (var parameter in testParametersDictionary)
+                            TestContext.Parameters.Add(parameter.Key, parameter.Value);
+                    }
+                }
+                else
+                {
```

Comment:
Just a suggestion - would this be clearer as an `if...else if` now the noise has been removed? I think that's me going well into the grounds of stylistic preference by now though - so feel free to ignore that. 😄